assets/js/youtube-api.js

```diff
@@ -13,22 +13,24 @@ class YoutubeAPI{
      * }]>}
      */
     static async getVideosMetadata(videosIDs){
-        let result = await fetch(`${GOOGLE_API_URL}videos?id=${encodeURIComponent(videosIDs.join(','))}&part=snippet,statistics&key=${API_KEY}`);
-        let resultJSON = await result.json()
-        return resultJSON.items.map(videoItem=>{
-            try{
-                return {
+        let result = await fetch(`${GOOGLE_API_URL}videos?id=${encodeURIComponent(videosIDs.join(','))}&part=snippet,statistics,status&key=${API_KEY}`);
+        let resultJSON = await result.json();
+        return resultJSON.items.reduce((acc, videoItem) => {
```

Comment:
I prefer using a map and a filter better than a reduce here

Reply:
Used reduce for performance (run through data once instead of twice)
But, data size `500` isn't that big. will revert to using filter + map